lib/inertia_rails/controller.rb

```diff
@@ -13,7 +13,7 @@ module Controller
       helper ::InertiaRails::Helper
 
       after_action do
-        cookies['XSRF-TOKEN'] = form_authenticity_token unless request.inertia? || !protect_against_forgery?
+        cookies['XSRF-TOKEN'] = form_authenticity_token unless !protect_against_forgery?
```

Comment:
Past discussions:
- https://github.com/inertiajs/inertia-rails/pull/96#discussion_r1242325750
- https://github.com/inertiajs/inertia-rails/pull/72 (referenced in the above)